prow/deploy.sh

```diff
@@ -0,0 +1,71 @@
+#!/bin/bash
+# Copyright 2016 The Kubernetes Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# bump.sh will
+# * ensure there are no pending changes
+# * optionally activate GOOGLE_APPLICATION_CREDENTIALS and configure-docker if set
+# * run //prow:release-push to build and push prow images
+# * update all the cluster/*.yaml files to use the new image tags
+
+set -o errexit
+set -o nounset
+set -o pipefail
+
+# TODO(fejta): rewrite this in a better language REAL SOON
+# TODO(fejta): update this and hack/print-workspace-status to customize STABLE_*_CLUSTER
+
+if [[ -n "${GOOGLE_APPLICATION_CREDENTIALS:-}" ]]; then
+  echo "Detected GOOGLE_APPLICATION_CREDENTIALS, activating..." >&2
+  gcloud auth activate-service-account --key-file="$GOOGLE_APPLICATION_CREDENTIALS"
+fi
+
+# See https://misc.flogisoft.com/bash/tip_colors_and_formatting
+
+color-context() { # Bold blue
+  echo -e "\x1B[1;34m${@}\x1B[0m"
+}
+
+color-missing() { # Light red
+  echo -e "\x1B[91m${@}\x1B[0m"
+}
+
+ensure-context() {
+  local proj=$1
+  local zone=$2
+  local cluster=$3
+  local context="gke_${proj}_${zone}_${cluster}"
+  echo -n "$(color-context "$context") "
+  kubectl config get-contexts "$context" &> /dev/null && return 0
+  shift
+  echo "$(color-missing MISSING)"
+  gcloud container clusters get-credentials --project="$proj" --zone="$zone" "$cluster"
+  kubectl config get-contexts "$context" > /dev/null
```

Comment:
I don't understand what is happening following line 50. Won't this always print "MISSING"? Why is `shift` needed?

Reply:
Done, removed unnecessary `shift`

If the context already exists we exit the function on L54 `kubectl get && return 0`